clippy_lints/src/functions/must_use.rs

```diff
@@ -117,38 +118,27 @@ fn check_needless_must_use(
                 fn_header_span,
                 "this unit-returning function has a `#[must_use]` attribute",
                 |diag| {
-                    diag.span_suggestion(attr.span, "remove the attribute", "", Applicability::MachineApplicable);
+                    diag.span_suggestion(
+                        attr.span(),
+                        "remove the attribute",
+                        "",
+                        Applicability::MachineApplicable,
+                    );
                 },
             );
         } else {
             // When there are multiple attributes, it is not sufficient to simply make `must_use` empty, see
             // issue #12320.
-            span_lint_and_then(
+            // FIXME(jdonszelmann): this used to give a machine-applicable fix. However, it was super fragile,
+            // honestly looked incorrect, and is a little hard to support for a little bit now. Some day this
+            // could be re-added.
+            span_lint_and_help(
                 cx,
                 MUST_USE_UNIT,
                 fn_header_span,
                 "this unit-returning function has a `#[must_use]` attribute",
-                |diag| {
-                    let mut attrs_without_must_use = attrs.to_vec();
-                    attrs_without_must_use.retain(|a| a.id != attr.id);
-                    let sugg_str = attrs_without_must_use
-                        .iter()
-                        .map(|a| {
-                            if a.value_str().is_none() {
-                                return a.name_or_empty().to_string();
-                            }
-                            format!("{} = \"{}\"", a.name_or_empty(), a.value_str().unwrap())
-                        })
-                        .collect::<Vec<_>>()
-                        .join(", ");
-
-                    diag.span_suggestion(
-                        attrs[0].span.with_hi(attrs[attrs.len() - 1].span.hi()),
-                        "change these attributes to",
-                        sugg_str,
-                        Applicability::MachineApplicable,
-                    );
-                },
```

Comment:
This suggestion was removed. I'd like a second pair of eyes on this.

Reply:
Not sure if you saw, but there is a comment above this 
```rs
            // FIXME(jdonszelmann): this used to give a machine-applicable fix. However, it was super fragile,
            // honestly looked incorrect, and is a little hard to support for a little bit now. Some day this could be
            // re-added.
```
so it sounds like it's temporarily being removed while the attributes are being refactored and intended to be re-added later?

Reply:
Yeah, I saw that. Just wanted a second pair of eyes on it. But if you also think it is ok to merge this, rather then immediately trying to fix it, we can merge :+1:

Reply:
(fyi) I've managed to construct examples since which this code for machine-application would completely mess up so in any case it was wrong. We can re-add it later (correctly :P)